presentation/src/main/java/com/mashup/alcoholfree/presentation/ui/component/SulSulWebViewClient.kt

```diff
@@ -0,0 +1,57 @@
+package com.mashup.alcoholfree.presentation.ui.component
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.util.Log
+import android.webkit.WebResourceRequest
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import java.net.URISyntaxException
+
+class SulSulWebViewClient(
```

Comment:
공유하기에만 사용되는 클래스인데 범용적인 이름을 갖는 것 보다는 SulSulWebView에 인자로 webViewClient를 넘겨주는 건 어때? 
nullable로 넘겨서 없으면 기본 클라이언트 쓰게 하면 좋을 것 같다는 생각이 드넹

Reply:
```suggestion
class KakaoLoginWebViewClient(
```

나도 지혜 의견에 완전히 동의해
공통 웹뷰에 클라이언트를 파라미터로 받게하고 얘는 로그인과 관련된 파람들을 받는게 좋다고 생각해

Reply:
움... 저는 사실 반대의견이긴한데 저희가 웹뷰들은 로딩 떄문에 무조건 onPageFinished 함수를 오버라이드 해야 된단 말이져? 
 
 그래서 기존 코드가 val client  = object: WebViewClient{ override ...} 와 비슷한 형태로 쓰이고 있었는데 이걸 분리 안하면고 웹뷰 클라이언트를 인자로 받을때 카카오 공유하기가 있는 웹뷰는 onPageFinished도 오버라이드하고, shouldOverrideUrlLoading도 오버라이드도 해서 지금 제가 작성한 SulSulWebViewClient 안의 저 긴코드가 작성된단 말이죠..?
 
어차피 null 체크로 카카오 공유하기가 없는 웹뷰는 shouldOverrideUrlLoading가 실행이 안되게 해놨고 클라이언트를 인자로 받아서 하자는 거는 움.. 이해를 못하겠습니당.. 코드로 보면 좀 다를순 있겠지만 제가 생각했을때는 지금 말씀 드린 의견과 같습니당...

Reply:
@oreocube 공유하기에만 사용되는 클래스가 아니라 모든 웹뷰에는 로딩 때문에 사용될것 같아~! 원래 그래서 카카오 공유하기에 있는 웹뷰만을 위한 웹뷰 컴포저블 함수를 만들었었는데 클라이언트에서 shouldOverrideUrlLoading 오버라이드 하는 거하고, 자바 인터페이스 설정에서 코드 한줄 추가하는 거 밖에는 차이가 없더라고.. 

그외에는 기존 웹뷰코드하고 동일했고 오히려 이거를 하나로 합치는게 낫다고 생각했어 그래서 지금 올린 코드처럼 되었구? 카카오톡 공유하기가 없는 웹뷰는 이미 null체크를 통해 실행이 안되게 해놨엉.

저 클라이언트를 분리를 안하면 지금 저 클래스에 있는 코드가 웹뷰 컴포저블 함수에 전부 들어가게 되더라고..! 오히려 코드가 지저분해져서 클래스로 분리해서 하는게 낫다고 생각했어~

Reply:
오홍..? 나도 클래스로 만든건 너무 좋아! 근데 내가 생각한건 이런 느낌이야

```kotlin
@Composable
fun SulSulWebView(
    modifier: Modifier = Modifier,
    url: String,
    isTransparent: Boolean,
    state: SulSulWebViewState? = null,
    bridge: SulSulWebViewBridge? = null,
    client: SulSulWebViewClient? = null,
    onIsWebViewLoading: (Boolean) -> Unit,
)

open class SulSulWebViewClient(
    private val onIsWebViewLoading: (Boolean) -> Unit,
) : WebViewClient() {

    override fun onPageFinished(view: WebView?, url: String?) {
        super.onPageFinished(view, url)
        onIsWebViewLoading(false)
    }
}

class KakaoShareWebViewClient(
    private val context: Context,
    private val onIsWebViewLoading: (Boolean) -> Unit,
    private val onStartKakao: (Intent) -> Unit,
) : SulSulWebViewClient(onIsWebViewLoading) {

    override fun shouldOverrideUrlLoading(
        view: WebView,
        request: WebResourceRequest?,
    ): Boolean { // ...
```

Reply:
아 이런의미였구나..? 이거면 인정하도록 하겠습니다..ㅎ 수정할겡~